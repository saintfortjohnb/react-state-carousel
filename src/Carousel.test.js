import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Carousel from "./Carousel";
import TEST_IMAGES from "./_testCommon.js";

it("works when you click on the right arrow", function() {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).toBeInTheDocument();
});

it("left arrow moves to previous image", function () {
  const { container } = render(
    <Carousel photos={TEST_IMAGES} title="Test Title" />
  );

  // Click right arrow to move to the second image
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);

  // Click left arrow to move back to the first image
  const leftArrow = container.querySelector(".bi-arrow-left-circle");
  fireEvent.click(leftArrow);

  // Expect the first image to be displayed again
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
});

it("left arrow is missing on first image, right arrow is missing on last image", function () {
  const { container } = render(
    <Carousel photos={TEST_IMAGES} title="Test Title" />
  );

  // Expect the left arrow to be missing on the first image
  const leftArrow = container.querySelector(".bi-arrow-left-circle");
  expect(leftArrow).toBeNull();

  // Click right arrow to move to the last image
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);
  fireEvent.click(rightArrow);

  // Expect the right arrow to be missing on the last image
  const updatedRightArrow = container.querySelector(".bi-arrow-right-circle");
  expect(updatedRightArrow).toBeNull();
});

it("renders without crashing", function () {
  render(<Carousel photos={TEST_IMAGES} title="Test Title" />);
});

it("matches snapshot", function () {
  const tree = renderer
    .create(<Carousel photos={TEST_IMAGES} title="Test Title" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
