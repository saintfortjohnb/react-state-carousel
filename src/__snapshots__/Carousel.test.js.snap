// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Carousel"
>
  <h1>
    Test Title
  </h1>
  <div
    className="Carousel-main"
  >
    <div
      className="Card"
    >
      <h4
        className="Card-title"
      >
        testing image 1
      </h4>
      <img
        alt="testing image 1"
        className="Card-image"
        src="test1.com"
      />
      <small
        className="Card-small"
      >
        Image 
        1
         of 
        3
        .
      </small>
    </div>
    <i
      className="bi bi-arrow-right-circle"
      onClick={[Function]}
    />
  </div>
</div>
`;
